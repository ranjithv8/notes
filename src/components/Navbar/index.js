import React from 'react';
import { Link } from 'react-router-dom';
const Navbar = ({ showHideNavBar, isOpen }) => (
  <nav className="navbar">
    <div className="container">
      <div className="navbar-brand">
        <Link className="navbar-item" to='/'>
          <p className='title is-2'>Notes</p>
        </Link>
        <span className={isOpen ? 'navbar-burger burger is-active' : 'navbar-burger burger'} onClick={showHideNavBar} data-target="navbarMenuHeroB">
          <span></span>
          <span></span>
          <span></span>
        </span>
      </div>
      <div id="navbarMenuHeroB" className={isOpen ? 'navbar-menu is-active' : 'navbar-menu'}>
        <div className="navbar-end">
          <span className="navbar-item">
            <Link className="button is-info is-inverted" to='/note/-1'>
              <span>Add Note</span>
            </Link>
          </span>
        </div>
      </div>
    </div>
  </nav>
)

export default Navbar;
/*
 <a className="navbar-item is-active">Home</a>
  <a className="navbar-item">Notes</a>
*/