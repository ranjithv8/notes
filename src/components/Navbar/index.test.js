import React from 'react';
import Navbar from './index';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
let isOpen = false;
const showHideNavBar = () => {
  isOpen = !isOpen;
}
test('Navbar class is toggled when the button is clicked', () => {
  let component = renderer.create(<MemoryRouter initialEntries={[ '/' ]}><Navbar showHideNavBar={showHideNavBar} isOpen={isOpen} /></MemoryRouter>);
  let tree = component.toJSON();
  let method = component.root.props.children.props.showHideNavBar;
  expect(tree).toMatchSnapshot();
  method();
  expect(tree).toMatchSnapshot();
  method();
  expect(tree).toMatchSnapshot();
});