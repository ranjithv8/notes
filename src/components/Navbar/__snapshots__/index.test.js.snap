// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar class is toggled when the button is clicked 1`] = `
<nav
  className="navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <p
          className="title is-2"
        >
          Notes
        </p>
      </a>
      <span
        className="navbar-burger burger"
        data-target="navbarMenuHeroB"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </span>
    </div>
    <div
      className="navbar-menu"
      id="navbarMenuHeroB"
    >
      <div
        className="navbar-end"
      >
        <span
          className="navbar-item"
        >
          <a
            className="button is-info is-inverted"
            href="/note/-1"
            onClick={[Function]}
          >
            <span>
              Add Note
            </span>
          </a>
        </span>
      </div>
    </div>
  </div>
</nav>
`;

exports[`Navbar class is toggled when the button is clicked 2`] = `
<nav
  className="navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <p
          className="title is-2"
        >
          Notes
        </p>
      </a>
      <span
        className="navbar-burger burger"
        data-target="navbarMenuHeroB"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </span>
    </div>
    <div
      className="navbar-menu"
      id="navbarMenuHeroB"
    >
      <div
        className="navbar-end"
      >
        <span
          className="navbar-item"
        >
          <a
            className="button is-info is-inverted"
            href="/note/-1"
            onClick={[Function]}
          >
            <span>
              Add Note
            </span>
          </a>
        </span>
      </div>
    </div>
  </div>
</nav>
`;

exports[`Navbar class is toggled when the button is clicked 3`] = `
<nav
  className="navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        <p
          className="title is-2"
        >
          Notes
        </p>
      </a>
      <span
        className="navbar-burger burger"
        data-target="navbarMenuHeroB"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </span>
    </div>
    <div
      className="navbar-menu"
      id="navbarMenuHeroB"
    >
      <div
        className="navbar-end"
      >
        <span
          className="navbar-item"
        >
          <a
            className="button is-info is-inverted"
            href="/note/-1"
            onClick={[Function]}
          >
            <span>
              Add Note
            </span>
          </a>
        </span>
      </div>
    </div>
  </div>
</nav>
`;
